### PR TITLE
Rango for Safari

### PR DESCRIPTION
**Pull request by nriley**
Wrap Rango as a Safari web extension. Implements clipboard support in the native app.

Update CONTRIBUTING.md with Safari instructions.

Here's the app's main window:

image

### CODE REVIEW

**Comment by david-tejada** [comment]
All seems okay (I guess 😅). I leave you the logos at different sizes in case you want to use those. If you think the ones you have are okay let me know and I can merge this one
[Rango logo.zip](https://github.com/david-tejada/rango/files/9102698/Rango.logo.zip)
.

**Comment by david-tejada** [comment]
I did some file reorganization and renaming. Now clipboard.ts is in `src/background/utils/clipboard.ts`. What's the best way to solve the conflict, you do a rebase?

**Comment by david-tejada** [comment]
I was trying to resolve the merge conflicts myself and I realized that you have to request "nativeMessaging" permission in Safari. I think we are going to need to create a different manifest file for Safari, because requesting permissions that you're not using could be problematic

**Comment by nriley** [comment]
> I was trying to resolve the merge conflicts myself and I realized that you have to request "nativeMessaging" permission in Safari. I think we are going to need to create a different manifest file for Safari, because requesting permissions that you're not using could be problematic

OK, whatever works, just let me know if you want me to do anything. Should have some time this weekend again.

**Comment by david-tejada** [comment]
I'll create another manifest and commands for building to dist-safari. I just need to know if there's anything else that needs to be changed from your code to reflect this change

**Comment by nriley** [comment]
> I'll create another manifest and commands for building to dist-safari. I just need to know if there's anything else that needs to be changed from your code to reflect this change 

Would just need to change the Xcode project file to point at the new manifest (and build folder if appropriate). Should be a 5 min change once you have the new manifest/build command done.

**Comment by nriley** [comment]
FYI, I am holding off on revising this PR until you let me know you're ready!

**Comment by david-tejada** [comment]
Yes, sorry. I'll do it this week so you can do your part on the weekend

**Comment by david-tejada** [comment]
I created the separate manifest and build scripts for Safari.

Btw, I bought an air m1 because the Linux machine I was using was very slow and was getting on my nerves. So next time I can try to package it myself for Safari

**Comment by nriley** [comment]
I've updated (finally) to the new manifest and undid my changes from the old one. May want to add a debugging guide, but here's some things to be aware of:

1) Don't ever have more than one copy of Rango.app on your system or results may be unpredictable (use Product > Clean Build Folder in Xcode if needed).
2) Some things that can help if Rango doesn't appear in Safari: uncheck/check in Safari Extensions Preferences; restart Safari.
3) Messages logged by the extension do not appear in Xcode but are logged to the system log (stream with `/usr/bin/log stream --predicate 'subsystem == "net.sabi.Rango.extension"' --info`).

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -42,3 +42,30 @@ _This example is for launching Brave on MacOS_
...
+### Safari
+
+Safari [only supports web extensions](https://developer.apple.com/documentation/safariservices/safari_web_extensions) bundled inside a native Mac app. If you are *not* building for development, Safari [only loads web extensions if their containing apps are distributed via the Mac App Store](https://developer.apple.com/documentation/safariservices/safari_web_extensions/distributing_your_safari_web_extension), or [if you select “Allow unsigned extensionsʺ from the Debug menu on each launch](https://developer.apple.com/documentation/safariservices/safari_web_extensions/running_your_safari_web_extension) (requires authentication).
+
+To build for development:
+
+1. Build the extension for manifest version 2:
+
+    ```bash
+    npm run build:mv2
```
Comment (david-tejada) [inline CONTRIBUTING.md]: I think this should be `npm run build:mv2-safari`.

Comment (nriley) [reply CONTRIBUTING.md]: Oops, fixed.

**Pull request merged**